pkg/log/log.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+// Factory builds Logger
+type Factory struct{}
+
+// Logger wraps logrus.Entry
+type Logger struct {
+	*logrus.Entry
+}
+
+// New sets logrus'output with lumberjack
+// filename: /path/to/log, e.g. "/var/log/sqlflow.log"
+func New(filename string) *Factory {
```

Comment:
Function `New` doesn't new an instance tho.
```go
package main

import "sqlflow.org/sqlflow/pkg/log"

func main() {
	logger1 := log.New("some_file")
	logger2 := log.New("some_other_file") // this overwrites the logger1
}
```

How about renaming `New` to `SetOutputFile`? So one can use it as
```go
package main

import "sqlflow.org/sqlflow/pkg/log"

func main() {
	log.SetOutputFile("some_file")
	log.WithFields(getGrpcSession).Info("some message")
}
```

Reply:
I prefer using  `log.SetOutput(io.Writer)`instead of `log.SetOutput(filename string)`, because in the workflow step, the logs should output to the stdout/stderr instead of file.

Reply:
> I prefer using log.SetOutput(io.Writer)instead of log.SetOutput(filename string), because in the workflow step, the logs should output to the stdout/stderr instead of file.

Good point. So we don't need to `import "gopkg.in/natefinch/lumberjack.v2"` in `pkg/log`.

Reply:
> > I prefer using log.SetOutput(io.Writer)instead of log.SetOutput(filename string), because in the workflow step, the logs should output to the stdout/stderr instead of file.
> 
> Good point. So we don't need `import "gopkg.in/natefinch/lumberjack.v2"` in `pkg/log`.

In fact, if we pass an empty string to `New`(OR replaced by SetOutputxXXX), the log content will be written to stdout.
So, let's take another look at this after the APIs(New&GetLogger) fixed.

Reply:
> Function `New` doesn't new an instance tho.
> 
> ```go
> package main
> 
> import "sqlflow.org/sqlflow/pkg/log"
> 
> func main() {
> 	logger1 := log.New("some_file")
> 	logger2 := log.New("some_other_file") // this overwrites the logger1
> }
> ```
> 
> How about renaming `New` to `SetOutputFile`? So one can use it as
> 
> ```go
> package main
> 
> import "sqlflow.org/sqlflow/pkg/log"
> 
> func main() {
> 	log.SetOutputFile("some_file")
> 	log.WithFields(getGrpcSession).Info("some message")
> }
> ```

You're right.
I intended to wrap the `logrus`, but it looks the user will be confused.